system/Entity.php

```diff
@@ -173,7 +188,21 @@ public function toArray(bool $onlyChanged = false, bool $cast = true): array
 			{
 				if (array_key_exists($to, $return))
 				{
-					$return[$from] = $this->__get($to);
+					$value = $this->__get($to);
```

Comment:
IMO using `$return[$key]` would be better. No need to introduce an additional variable.